pages/Identity/Registry.md

```diff
@@ -0,0 +1,154 @@
+---
+name: Registry
+route: /Identity/Registry
+menu: Identity
+---
+
+# Identity Registry
+
+A single contract is required for uniqueness of handles across the DSNP.
+Additionally the registry allows for shorter ids in a graph
+and allows switching identity contracts while maintaining all graph connections, public and private.       
+
+## Specification Status
+
+| Version | Status |
+---------- | ---------
+| 0.1     | Draft |
+
+## Purpose
+1. Describes how an Identity Registry resolves a name or number to an identity contract address
+
+## Assumptions
+
+* Handles will need to be moved from Betanet to Mainnet 
+
+## Discovery
+
+Contract addresses are not easy to remember and while some networks have had success with just numbers,
+it is most common to think of text based handles for finding other people on a network.
+
+## Graph Connections
+
+Ethereum contract addresses are currently 160 bit values which is much larger than needed for unique identification.
+Graph connections can be reduced to just 64 bit identifiers.
+
+## EIP 721
+
+This needs to be evaluated and discussed for inclusion/exclusion.
+
+While the NFT standard is not necessary to support, it provides standard ownership transfer and management tools.
+
+## Rejected Alternatives
+
+* Why not the Ethereum Name Service ([ENS](https://ens.domains/))?
+  * Overly complex and expensive for the simple use case of handles
+  * Support can be added later through a custom ENS resolver
+  * Didn't provide a numerical id for efficient graph storage
+
+
+## Discussion Points
```

Comment:
Discussion in a meeting:

* Should the interface support reverse resolutions? (id -> handle, address -> id/handle)
  * YES, it is just better to be able to reverse
* Should we still match the ENS Resolver spec [EIP 137](https://eips.ethereum.org/EIPS/eip-137#resolver-specification)?
  * NO, stay away so that the future can implement it if wanted
* Should we name this registry or resolver or split?
  * NO, keep it together as there is no need to follow the ENS style split
* Do we need the uint64 id or should we just use the handle?
  * uint64 would be really nice, let's try to keep it and adjust if the implementation makes it difficult.
* Are we going to hash the handle (ENS style) client side?
  * NO, we will store the strings (required for reverse resolutions)
* Can more than one handle point to the same social identity?
  * NO. While there are some ways around this, the registry does not allow for it. It would make name squatting easier and.
* Are we case sensitive / normalizing?
  * Strings are UTF-8
  * Add a new story to investigate and a note that it is an open question
* Do we need a way to have names expire or be garbage collected?
  * Maybe? Future discussions.
* Can we represent ownership of a name without having to have a contract behind that name?
  * Yes we could potentially allow for wallets to own a name, but that would not then resolve to a DSNP Identity, but it raises other questions:
    * Would we allow that key to sign messages?
    * Would we test against the interface to check?
    * Would this allow for too much flexibility?